flipperdevices/flipperzero-firmware
# fbt: Fix building using path with space

# What's new

Fix building (on Windows?) using path with space (fixes #2947)

Also, many variables are already typed with `FS.File`/`FS.Dir` type, and `ASSETS_COMPILER` and `NANOPB_COMPILER` are concatenated with `FS.Dir` paths, so no extra quoting needed. `ICON_FILE_NAME` on the other hand is a str.

To "manually debug", SCons/Action.py, after `cmd_list` definition (~line 964):
```py
        print('TOTO', cmd_list)
        raise SystemExit()
```

Possible alternatives for `extract_abs_dir_path()`:
```py
return SCons.Subst.Literal(abs_dir_node.abspath)
return SCons.Subst.SpecialAttrWrapper(abs_dir_node.abspath)
```

# Verification 

- Use the patched build environment on a known working copy (regression check)
- Use the known working build environment copied on a path containing spaces (see reproduction in linked issue)

# Checklist (For Reviewer)
